src/main/java/co/rsk/bitcoinj/core/StoredBlock.java

```diff
@@ -113,13 +120,33 @@ public StoredBlock getPrev(BtcBlockStore store) throws BlockStoreException {
         return store.get(getHeader().getPrevBlockHash());
     }
 
-    /** Serializes the stored block to a custom packed format. Used by {@link CheckpointManager}. */
-    public void serializeCompact(ByteBuffer buffer) {
-        byte[] chainWorkBytes = Utils.bigIntegerToBytes(getChainWork(), CHAIN_WORK_BYTES);
-        if (chainWorkBytes.length < CHAIN_WORK_BYTES) {
-            // Pad to the right size.
-            buffer.put(EMPTY_BYTES, 0, CHAIN_WORK_BYTES - chainWorkBytes.length);
-        }
+    /**
+     * @deprecated Use {@link #serializeCompactV2(ByteBuffer)} instead.
+     *
+     * Serializes the stored block to a custom packed format. Used internally.
+     * As of June 22, 2024, it takes 12 unsigned bytes to store the chain work value,
+     * so developers should use the V2 format.
+     *
```

Comment:
We should probably add the @Deprecated tag and explain that the new method should be called instead of this one

Reply:
It already has it. Not sure if I'm following you well.